src/aktualizr_info/main.cc

```diff
@@ -73,122 +81,66 @@ int main(int argc, char **argv) {
-    if (!storage->loadEcuSerials(&serials)) {
-      std::cout << "Couldn't load ECU serials" << std::endl;
-    } else if (serials.size() == 0) {
-      std::cout << "Primary serial is not found" << std::endl;
-    } else {
-      std::cout << "Primary ecu serial ID: " << serials[0].first << std::endl;
-      std::cout << "Primary ecu hardware ID: " << serials[0].second << std::endl;
-    }
-    if (serials.size() > 1) {
-      auto it = serials.begin() + 1;
-      std::cout << "Secondaries:\n";
-      int secondary_number = 1;
-      for (; it != serials.end(); ++it) {
-        std::cout << secondary_number++ << ") serial ID: " << it->first << std::endl;
-        std::cout << "   hardware ID: " << it->second << std::endl;
-
-        std::vector<Uptane::Target> installed_targets;
-        size_t current_version = SIZE_MAX;
-        size_t pending_version = SIZE_MAX;
-
-        auto load_installed_version_res = storage->loadInstalledVersions((it->first).ToString(), &installed_targets,
-                                                                         &current_version, &pending_version);
-
-        if (!load_installed_version_res ||
-            (current_version >= installed_targets.size() && pending_version >= installed_targets.size())) {
-          std::cout << "   no details about installed nor pending images\n";
-        } else {
-          if (installed_targets.size() > current_version) {
-            std::cout << "   installed image hash: " << installed_targets[current_version].sha256Hash() << "\n";
-            std::cout << "   installed image filename: " << installed_targets[current_version].filename() << "\n";
-          }
-          if (installed_targets.size() > pending_version) {
-            std::cout << "   pending image hash: " << installed_targets[pending_version].sha256Hash() << "\n";
-            std::cout << "   pending image filename: " << installed_targets[pending_version].filename() << "\n";
-          }
-        }
-      }
+      return EXIT_FAILURE;
     }
 
-    std::vector<MisconfiguredEcu> misconfigured_ecus;
-    storage->loadMisconfiguredEcus(&misconfigured_ecus);
-    if (misconfigured_ecus.size() != 0u) {
-      std::cout << "Removed or not registered ecus:" << std::endl;
-      std::vector<MisconfiguredEcu>::const_iterator it;
-      for (it = misconfigured_ecus.begin(); it != misconfigured_ecus.end(); ++it) {
-        std::cout << "   '" << it->serial << "' with hardware_id '" << it->hardware_id << "' "
-                  << (it->state == EcuState::kOld ? "has been removed from config" : "not registered yet") << std::endl;
-      }
+    if (vm.count("name-only") != 0u) {
+      std::cout << device_id << std::endl;
+      return EXIT_SUCCESS;
     }
 
-    auto registered = storage->loadEcuRegistered();
-    std::cout << "Provisioned on server: " << (registered ? "yes" : "no") << std::endl;
-    std::cout << "Fetched metadata: " << (has_metadata ? "yes" : "no") << std::endl;
-
     if (has_metadata) {
```

Comment:
Maybe it was already the case before, but if the user specified one of the following options and `has_metadata` is false, only the generic help would be displayed which won't be too useful for diagnostic. It should probably fail and display an appropriate error message.

Reply:
Is it mean that we should consider the case when the user chose an argument which depends on metadata and shows error message in case the has_metadat=false. 
In that case, the flow will be:
If the user chose the option which depends on has_metadata and has_metadata=false we print general info and application terminate with error code. 
In this case, the --info option will not be necessary.

Reply:
> If the user chose the option which depends on has_metadata and has_metadata=false we print general info and application terminate with error code.

Yes that's what I meant. Though, I'm not sure why it would display general info if called with `--image-targets` (for example)? Maybe an error message is enough?

Reply:
Yes I agree with you, if you look into the code we have "if" statement if(ha_metadata) {...}
I think we can remove that and check the has_metadata variable individually for each argument which is related to it, 
for example:
if (vm.count("images-root") != 0u) {
  if (!has_metadata) {
     // print message
     return EXIT_FAILURE;
  }
  ...
}

The general info will be printed in case the user leaves arguments empty.

Reply:
yes, sounds good!

Reply:
One question regarding Delegations, currently the aktualizr info test is designed when it expects that process will terminate always with EXIT_SUCCESS status, but actually, it's not always true, since we going to run the app as a child process we can analyze status code EXIT_SUCCESS\EXIT_FAILURE, and based on the exit status decide what we should do with stdout.
For that reason, I guess it makes sense to check exit status for each particular case.
The main idea that we must not analyze the STDOUT but the status code.
In the case of "Delegations", currently, app return EXIT_SUCESS and the message "Delegations are not present".
https://github.com/advancedtelematic/aktualizr/blob/d69f4f60d5ecc26de3ccb30a702bac693db50e54/src/aktualizr_info/aktualizr_info_test.cc#L56

Reply:
No delegations is a valid (successful) state, so if we are testing the case where there are delegations, we have to check stdout to verify that. @mike-sul wrote that IIRC and can comment if I'm mistaken.

Reply:
Ok, I understand, (no delegation it is a valid state),  
the (EXIT_SUCCESS\EXIT_FAILURE) is actually nothing more than just 0/1, and we can interpret this as (IS_DELEGATION/NO_DELEGATION).

What it gives us!
In the previous version, if there are no delegations, we can know about that only comparing string from stdout: 
if stdout == "Delegations are not present"

With  different exit code we can:
if exit code == 1

And then only if we really want to deal with that and check what happens, we can compare string fropm stdout.

Let me know if it makes sense.

Reply:
> we can interpret this as (IS_DELEGATION/NO_DELEGATION).

Actually, I don't think so. `EXIT_FAILURE` should only be used if there was a problem accessing the database or parsing its contents (or something similarly erroneous). `EXIT_SUCCESS` is returned whether or not there are delegations present.

>  if exit code == 1

You are correct that generally `EXIT_SUCCESS` and `EXIT_FAILURE` normally resolve to 0 and 1, but that is not guaranteed, which is why we recommend using the provided variables for comparison.

Reply:
Ok, it's clear, thanks for the clarification, could you also check other cases with exit FAILURE, in particular, loading root metadata?

Reply:
I think you can see that from the source code of aktualizr-info's main.cc. The only time `EXIT_FAILURE` occurs is when there is some significant problem, like inability to read the database.